tracer.go

```diff
@@ -23,4 +23,21 @@ type Tracer interface {
 	//         SetTag("lucky_number", 42)
 	//
 	StartTrace(operationName string) Span
+
+	// ImplementationID returns information about the OpenTracing
+	// implementation backing the Tracer. The return value should not change
+	// over the life of a particular Tracer instance.
+	ImplementationID() *ImplementationID
+}
+
+// ImplementationID is a simple, extensible struct that describes an
+// OpenTracing implementation.
+type ImplementationID struct {
+	// The (stable) name of the implementation. E.g., "Dapper" or "Zipkin". The
+	// Name should not reflect the host language or platform.
+	Name string
+
+	// Version may take any form, but SemVer (http://semver.org/) is strongly
+	// encouraged.
+	Version string
```

Comment:
@bensigelman , this looks pretty good, but Go seems a bit like the exceptional case with regards to the utility of a version string (since everything in Go is basically statically bound). I'm curious what this might look like for Python. I'm specifically considering the case where the *supported version of the OpenTracing standard* is what's of interest rather than the underlying *implementation version*. E.g. if an arbitrary piece of Python code is handed a `tracer` object, how can it (cleanly) check the supported version of OpenTracing on that object?

(By the way I believe I'm safe in assuming that in the above `Version` is intended to represent the implementation version given the struct name and the "may take any form" comment.)